KeyDistribution_BB84/ReferenceImplementation.qs

```diff
@@ -90,17 +90,17 @@ namespace Quantum.Kata.KeyDistribution {
     }
 
 
-    // Task 2.5. Was communication secure?
-    function CheckKeysMatch_Reference (keyAlice : Bool[], keyBob : Bool[], threshold : Int) : Bool {
+    // Task 2.5. Check if error rate was low enough
+    function CheckKeysMatch_Reference (keyAlice : Bool[], keyBob : Bool[], errorRate : Int) : Bool {
         let N = Length(keyAlice);
         mutable count = 0;
         for (i in 0 .. N - 1) {
-            if (keyAlice[i] == keyBob[i]) {
+            if (keyAlice[i] != keyBob[i]) {
                 set count += 1;
             }
         }
 
-        return IntAsDouble(count) / IntAsDouble(N) >= IntAsDouble(threshold) / 100.0;
+        return IntAsDouble(count) / IntAsDouble(N) >= IntAsDouble(errorRate) / 100.0;
```

Comment:
I think this condition has to be updated as well, if we're asking to return ""true" if the percentage of errors is **less than or equal** to the error rate"?

```suggestion
        return IntAsDouble(count) / IntAsDouble(N) <= IntAsDouble(errorRate) / 100.0;
```